### PR TITLE
Remove dead code from the VM.

### LINKED ISSUE #1598: Remove dead code from the VM.
**Issue summary**

The VM structure contains a bunch of dead code from before the switch to the FVM. It should be removed together with as many of the lifetime variables as possible:

https://github.com/ChainSafe/forest/blob/df5227043f9f72c1a2b5e8f552b9bc8490b178c1/vm/interpreter/src/vm.rs#L94-L117

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1598

**Other information and links**

### CODE REVIEW

File: vm/interpreter/src/vm.rs [hunk 1]
```diff
@@ -155,7 +133,12 @@ where
         lb_state: &'r LB,
         engine: Engine,
         heights: Heights,
-    ) -> Result<Self, String> {
+    ) -> Result<Self, String>
+    where
+        R: Rand + Clone + 'static,
+        C: CircSupplyCalc,
+        LB: LookbackStateGetter<'db, DB>,
```
Comment (lemmih) [inline vm/interpreter/src/vm.rs:140]: Can `'db` be changed to `'_`?

File: vm/interpreter/src/vm.rs [hunk 2]
```diff
@@ -155,7 +133,12 @@ where
         lb_state: &'r LB,
```
[the remark this answers is not in the document]
Comment (tyshko5) [reply vm/interpreter/src/vm.rs]: Removed

File: vm/interpreter/src/vm.rs [hunk 3]
```diff
@@ -132,30 +113,32 @@ pub fn import_actors(blockstore: &impl BlockStore) -> BTreeMap<NetworkVersion, C
...
         store_arc: Arc<DB>,
         epoch: ChainEpoch,
-        rand: &'r R,
+        rand: &R,
         base_fee: BigInt,
         network_version: NetworkVersion,
         circ_supply_calc: C,
         override_circ_supply: Option<TokenAmount>,
-        lb_state: &'r LB,
+        lb_state: &'db LB,
```
Comment (lemmih) [inline vm/interpreter/src/vm.rs:133]:
```suggestion
        lb_state: &LB,
```

**Review by lemmih — CHANGES_REQUESTED** [review]

File: vm/interpreter/src/vm.rs (hunk 3 above)
Comment (tyshko5) [reply vm/interpreter/src/vm.rs]: Fixed

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**